spec/requests/groups_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe 'Groups', type: :request do
+  include Devise::Test::IntegrationHelpers
+
+  let(:user) { User.create(name: 'User1', email: 'Test@mail.com', password: '123456') }
+  let(:group) { user.groups.create!(name: 'shopping', icon: 'img1.png') }
+
+  describe 'GET /index' do
+    before do
+      sign_in user
+      get groups_path
+    end
+
+    it 'should return response status correct (ok)' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'respons to html' do
+      expect(response.content_type).to include 'text/html'
+    end
+  end
+
+  describe 'GET /show' do
+    before do
+      sign_in user
+      get group_path(1)
+    end
+    it 'should return response status correct (ok)' do
+      expect(response).to have_http_status(:ok)
+    end
+    it 'respons to html' do
+      expect(response.content_type).to include 'text/html'
+    end
+    it 'should include correct name of group' do
+      expect(response.body).to include('shopping')
+    end
+  end
+
+  describe 'GET /new' do
+    before do
+      sign_in user
+      get new_group_path
+    end
```

Comment:
- [ ] Some of the tests are failing. Please make sure to fix those. :slightly_smiling_face: 

![Screenshot from 2022-03-16 21-59-46](https://user-images.githubusercontent.com/22434660/158640937-5b3f0477-cc2d-4c04-b552-ba74b27eac9a.png)

Reply:
They was working with but I will check them again.